src/Core/ExecutionPathTracer/ExecutionPath.cs

```diff
@@ -0,0 +1,131 @@
+        /// </summary>
+        /// <param name="qubits">
+        /// An array of <c>QubitDeclaration</c>s that represents the declared qubits used in the execution path.
+        /// </param>
+        /// <param name="operations">
+        /// An array of <c>Operation</c>s that represents the operations used in the execution path.
+        /// </param>
+        public ExecutionPath(QubitDeclaration[] qubits, Operation[] operations)
+        {
+            this.Qubits = qubits;
+            this.Operations = operations;
+        }
+
+        /// <summary>
+        /// An array of <c>QubitDeclaration</c>s that represents the declared qubits used in the execution path.
+        /// </summary>
+        public QubitDeclaration[] Qubits { get; set; }
+
+        /// <summary>
+        /// An array of <c>Operation</c>s that represents the operations used in the execution path.
+        /// </summary>
+        public Operation[] Operations { get; set; }
+
+        /// <summary>
+        /// Serializes <c>ExecutionPath</c> into its JSON representation.
+        /// </summary>
+        /// <param name="prettyPrint">
+        /// Pretty prints the JSON (i.e. with white space and indents) if true (default = false).
+        /// </param>
+        public string ToJson(bool prettyPrint = false)
+        {
+            var options = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                IgnoreNullValues = true,
+                WriteIndented = prettyPrint,
+            };
+            return JsonSerializer.Serialize(this, options);
+        }
+    }
+
+    /// <summary>
+    /// Represents a qubit resource used in an execution path.
+    /// </summary>
+    public class QubitDeclaration
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QubitDeclaration"/> class.
+        /// </summary>
+        /// <param name="id">
+        /// Id of qubit.
+        /// </param>
+        /// <param name="numChildren">
+        /// Number of associated classical registers.
+        /// </param>
+        public QubitDeclaration(int id, int? numChildren = null)
+        {
+            this.Id = id;
+            this.NumChildren = numChildren;
+        }
+
+        /// <summary>
+        /// Id of qubit.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Number of associated classical registers.
+        /// </summary>
+        public int? NumChildren { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an operation used in an execution path.
+    /// </summary>
+    public class Operation
+    {
+        /// <summary>
+        /// Label of gate.
+        /// </summary>
+        public string Gate { get; set; }
+
+        /// <summary>
+        /// Non-qubit arguments provided to gate.
+        /// </summary>
+        public string ArgStr { get; set; }
```

Comment:
We only want to render the non-qubit arguments in the argument string when we visualize them as the qubits they act on are already implied by the visualization. Would `Arguments` be a better property name?

Reply:
I'm a bit mixed here, as there's some potential for confusion with Q# syntax. That is, qubits are just ordinary values of type `Qubit`, such that the input tuple would look confusing if you just pulled out the qubits. For example, how would this property represent an input of type [`FixedPoint`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.arithmetic.fixedpoint), since that UDT is declared as having `(Int, Qubit[])` as its base type?

Reply:
That would most likely (if represented as a tuple type) would return `(Int)`. So, if we have `Foo(FixedPoint)` the visualization would create a gate acting on `Qubit[]` wires and the label `Foo(Int)`.

Reply:
Changed the name to `DisplayArgs` to denote that these might not contain all arguments, only the arguments we want to display.